### ISSUE TITLE
stop_gradient for alpha?

### ISSUE THREAD

**Issue by jonkoi** [XJTUWYD/TWN#1]
Hi,

I think there should be a tf.stop_gradient in compute_alpha so that the gradients do not pass through it and directly update weights. I haven't checked if any of the other ops in the function cut off gradients like tf.sign though.
